src/test/java/com/connexta/ingest/IngestITests.java

```diff
@@ -54,47 +135,99 @@
   @Value("${endpoints.transform.version}")
   private String endpointsTransformVersion;
 
-  @Inject private MockMvc mvc;
-  @Inject private RestTemplate restTemplate;
+  @Value("${s3.bucket}")
+  private String s3Bucket;
+
+  @Inject private AmazonS3 amazonS3;
+
+  @Inject
+  @Named("storeClientRestTemplate")
+  private RestTemplate storeClientRestTemplate;
 
   @Inject
-  @Named("nonBufferingRestTemplate")
-  private RestTemplate nonBufferingRestTemplate;
+  @Named("transformClientRestTemplate")
+  private RestTemplate transformClientRestTemplate;
 
-  private MockRestServiceServer storeServer;
-  private MockRestServiceServer transformServer;
+  private MockRestServiceServer storeMockRestServiceServer;
+  private MockRestServiceServer transformMockRestServiceServer;
+
+  @Inject private WebTestClient webTestClient;
+
+  @TestConfiguration
+  static class Config {
+
+    @Bean
+    public AmazonS3Configuration testAmazonS3Configuration() {
+      return new AmazonS3Configuration(
+          String.format(
```

Comment:
:pencil2:  Avoid using string templates to build URIs/URLs. Use a URI/URL constructor a builder like UriBuilder or UriComponentsBuilder to create an object. Specialized classes offer better safety and a richer interface than strings.

Reply:
https://github.com/connexta/ion-ingest/pull/23#discussion_r332773878